os/fs/driver/mtd/smart.c

```diff
@@ -5531,12 +5539,20 @@ static int smart_journal_scan(FAR struct smart_struct_s *dev, bool print_dump)
 		fdbg("Journal Scan Start !!\n");
 	}
 	dev->journal_seq = 0;
+
+	dev->block_map = (FAR uint16_t *)kmm_zalloc(dev->njournaleraseblocks * sizeof(uint16_t));
+	if (dev->block_map == NULL) {
+		fdbg("out of memory!!\n");
+		return -ENOMEM;
+	}
+	
 	for (dev->journal_seq = 0; dev->journal_seq < dev->njournalentries; dev->journal_seq++) {
 		ret = smart_journal_read_journal_log(dev, &log);
 		if (ret < 0) {
 			fdbg("Reading Journal Entry Failed! ret : %d journal_seq : %d\n", ret, dev->journal_seq);
 			return -EIO;
```

Comment:
Is it okay not to free the `dev->block_map` for failure case?

Reply:
yes it will free when smartfs unmounted